flink-clients/src/main/java/org/apache/flink/client/deployment/ClusterDescriptor.java

```diff
@@ -44,5 +45,12 @@
 	 * @return Client for the cluster
 	 * @throws UnsupportedOperationException if this cluster descriptor doesn't support the operation
 	 */
-	ClientType deploy() throws UnsupportedOperationException;
+	ClientType deploySession() throws UnsupportedOperationException;
```

Comment:
Would it make sense to rename these to `deployForSession` and `deployForJob`? Judgint from the names it seems reasonable to call `deployJob` after `deploySession`.